### PR TITLE
fix(eslint-plugin): [no-unnecessary-template-expression] should underline template syntax with squiggly lines

### LINKED ISSUE #8597: Bug: [no-unnecessary-template-expression] Report squigglies should underline template syntax, not just expression inside
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&showAST=es&fileType=.tsx&code=AYJwpgDg9iAuAEBnAjgVwJYHNMBt1kXgBIBvAKHngHor4ATKeAOygVSbrBDybHlgAW6QgGMoAW3FgmsADTwY-AX1IKmSvhHAA3dFFSEeYAHQV4AIgAqygJ5qcd9p27peS4UlghXmeQDMQCXg0Vj5YRhDYE3MzGnhrPgBDVHC-dAAPeAB3dBwcT28IBRSNRD4xSWlYQkSODWISeFq6eABfeURGAEl3JgBreHQERAF9HBbY2icuIw1xJsIssDz5cPgRRLym2CiZPXU1wT4s0Zw%2BVyiQaBxE2H34ACMwH0n6rSgHs-F5FgQjwaYvBA8DA6S0BEQ9yGZRwflMrWAQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS18iALbF4AQ2G1Gw6BPioMkRNGgdokcGAC%2BIHUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
`report squigglies ${
  // do not underline this comment, or the ${ on the previous line.
  "They only underline this string, from quote to quote."
  // The autofix will strip out these comments and the ${ and }, so I think it should 
  // underline them as well, to call attention to the whole interpolation being
  // the problem, not the inner expression itself.
}`
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-useless-template-literals": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Expect error squigglies starting on the `${` and continuing all the way through the closing `}`

### Actual Result

only the quoted string itself has squigglies

### Additional Info

I was playing around with this, and it's a very easy change to make. Would be happy to toss up a PR.

### PR DESCRIPTION
**Pull request by ronami**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8597
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

This PR addresses #8597 and changes the squiggles marked by `@typescript-eslint/no-unnecessary-template-expression` from the following:

```ts
`use${'less'}template${'expressions'}`
      ^^^^^^           ^^^^^^^^^^^^^
```

To also include the template syntax instead:

```ts
`use${'less'}template${'expressions'}`
    ^^^^^^^^^        ^^^^^^^^^^^^^^^^
```

### CODE REVIEW

**Review by kirkwaiblinger — DISMISSED** [review]
Lovely! And congrats on your first time contributing a PR to typescript-eslint ❤️ 

thumbs up gif

File: packages/eslint-plugin/src/rules/no-unnecessary-template-expression.ts [hunk 1]
```diff
@@ -106,6 +107,10 @@ export default createRule<[], MessageId>({
         if (hasSingleStringVariable) {
           context.report({
             node: node.expressions[0],
```
Comment (kirkwaiblinger) [inline packages/eslint-plugin/src/rules/no-unnecessary-template-expression.ts:109]: (optional) note that `node` and `loc` are redundant (with `loc` taking precedence), so we can leave `node` off in both these reports.

**Review by JoshuaKGoldberg — DISMISSED** [review]
ditto 😄

File: packages/eslint-plugin/src/rules/no-unnecessary-template-expression.ts (hunk 1 above)
Comment (kirkwaiblinger) [reply packages/eslint-plugin/src/rules/no-unnecessary-template-expression.ts:109]: ie,
```suggestion
```

Comment (ronami) [reply packages/eslint-plugin/src/rules/no-unnecessary-template-expression.ts]: Cool, wasn't aware of that, thanks :)

File: packages/eslint-plugin/src/rules/no-unnecessary-template-expression.ts [hunk 2]
```diff
@@ -245,20 +249,18 @@ export default createRule<[], MessageId>({
             ]);
           }
 
+          const warnLocStart = prevQuasi.range[1] - 2;
+          const warnLocEnd = nextQuasi.range[0] + 1;
+
           context.report({
             node: expression,
+            loc: rangeToLoc(context.sourceCode, [warnLocStart, warnLocEnd]),
```
Comment (kirkwaiblinger) [inline packages/eslint-plugin/src/rules/no-unnecessary-template-expression.ts:257]: this one too then 🙂 
```suggestion
          context.report({
            loc: rangeToLoc(context.sourceCode, [warnLocStart, warnLocEnd]),
```

**Review by kirkwaiblinger — APPROVED** [review]

**Review by auvred — APPROVED** [review]
:+1:

**Pull request merged**